mongo/integration/unified/client_entity.go

```diff
@@ -80,6 +90,16 @@ func newClientEntity(ctx context.Context, em *EntityMap, entityOptions *entityOp
 		}
 	}
 
+	// TODO: add explanation
+	if olm := entityOptions.ObserveLogMessages; olm != nil {
+		// We buffer the logQueue to avoid blocking the logger goroutine.
+		entity.logQueue = make(chan orderedLogMessage, clientEntityLogQueueSize)
+
+		if err := setLoggerClientOptions(entity, clientOpts, olm); err != nil {
+			return nil, fmt.Errorf("error setting logger options: %v", err)
```

Comment:
Use the `%w` format verb instead of `%v` so `fmt.Errorf` creates a wrapped error.